### PR TITLE
clean up code around setting geth client headers

### PR DESCRIPTION
**Pull request by roberto-bayardo**
**What type of PR is this?**

Cleanup

**What does this PR do? Why is it needed?**

Uses the more idiomatic gethClient.WithHeaders method for setting http headers for the geth client.

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

cc: @terencechain

### CODE REVIEW

**Comment by roberto-bayardo** [comment]
Leaving this as draft for now since I would like to get rid of the replace directive I had to add to go.mod. I'll be able to do this once https://github.com/MariusVanDerWijden/tx-fuzz/pull/9 is merged.  (cc: @MariusVanDerWijden)

File: go.mod [hunk 1]
```diff
@@ -1,21 +1,21 @@
 module github.com/prysmaticlabs/prysm/v3
 
-go 1.19
+go 1.18
```
Comment (prestonvanloon) [inline go.mod:3]: Any reason to downgrade?

Comment (roberto-bayardo) [reply go.mod:3]: apologies no, this is just for the machine I was testing this on (need to update the go binary!)  I will revert this once I can also remote the replace directive.

**Comment by roberto-bayardo** [comment]
So turns out geth v1.10.26 doesn't actually contain commit 90711efb0ab6360c37ab2135ab961daf05e83a74 which this change relies on, so we'll still need to keep the replace directive in here for prysm to build.   Is that OK?

File: go.mod (hunk 1 above)
Comment (roberto-bayardo) [reply go.mod]: fixed

File: go.mod [hunk 2]
```diff
@@ -258,3 +258,5 @@ replace github.com/MariusVanDerWijden/tx-fuzz => github.com/marcopolo/tx-fuzz v0
 
 // See https://github.com/prysmaticlabs/grpc-gateway/issues/2
 replace github.com/grpc-ecosystem/grpc-gateway/v2 => github.com/prysmaticlabs/grpc-gateway/v2 v2.3.1-0.20220721162526-0d1c40b5f064
+
+replace github.com/ethereum/go-ethereum => github.com/ethereum/go-ethereum v1.10.24-0.20221209152046-3775e198dfc9 // required until commit 90711efb0ab6360c37ab2135ab961daf05e83a74 is included in a tagged release
```
Comment (prestonvanloon) [inline go.mod:262]: Can't we add go-ethereum like this if we need https://github.com/ethereum/go-ethereum/commit/90711efb0ab6360c37ab2135ab961daf05e83a74?
```
go get github.com/ethereum/go-ethereum@90711efb0ab6360c37ab2135ab961daf05e83a74
```

Comment (roberto-bayardo) [reply go.mod:262]: Disclaimer: I am a go mod n00b

When I tried to go-get that specific commit, a bunch of dependencies get downgraded by go mod tidy and then it complains about ambiguous imports.

if I try to go-get a more recent geth commit, then I get:

```
[prysm]: go get github.com/ethereum/go-ethereum@3775e198dfc9fec312f44fd563dcc525cefa1e5d
go: github.com/ethereum/go-ethereum@3775e198dfc9fec312f44fd563dcc525cefa1e5d (v1.10.24-0.20221209152046-3775e198dfc9) requires github.com/ethereum/go-ethereum@v1.10.26, not github.com/ethereum/go-ethereum@3775e198dfc9fec312f44fd563dcc525cefa1e5d (v1.10.24-0.20221209152046-3775e198dfc9)
```

**Comment by MariusVanDerWijden** [comment]
I merged the pr btw

**Comment by roberto-bayardo** [comment]
> Problem is that we started splitting the head between master and v.24 (I think) v.25 and v.26 are backports while master is running further completely unrelated

Right, so like a v1.20.27-alpha or similar tag on the master branch some time after 1.10.24 could possibly allow us to tag a master commit without gomod overriding it with v.26?

**Comment by rauljordan** [comment]
hi @roberto-bayardo is this still relevant? Would be great to get this checked in soon if so

**Comment by terencechain** [comment]
I think this may be a blocker for Deneb

**Comment by roberto-bayardo** [comment]
Thanks for pinging I had totally lost track of this.  I have rebased the repo and the good news is the annoying go.mod directive I had to add is no longer needed.  I think it's still relevant if one wants an authenticated connection with geth. I'm a little surprised it wasn't needed for shapella though. @terencechain is the authenticated connection a deneb requirement?

**Comment by roberto-bayardo** [comment]
I cleaned up the change a little: there's no point in calling "SetHeader" after the switch at all since we now provide the headers in DialOptions for http clients, and it is a no-op for non-http clients.

**Comment by roberto-bayardo** [comment]
I did some more digging and indeed previous commits had fixed the core issue already.  I updated this PR to just a simple cleanup that makes the header initialization a bit more idiomatic.  No longer a critical change though.

**Review by rauljordan — APPROVED** [review]

**Review by OmnisAetas — APPROVED** [review]

**Comment by rkapka** [comment]
Hi @roberto-bayardo , we would like to merge the PR, but it has conflicts. Can you please resolve them?

**Pull request merged**